### DISCUSSION TITLE
Modelling of POS and morphological features

### DISCUSSION THREAD

**Discussion by gcelano** [korpling/ANNIS#871] [General]
In my corpus, POS + morphological features are encoded as a 9-character string, with the first character being the POS and the others the morphological features. If I encode this string as it is in PAULA, and then write a ANNIS query such as `postag=/v.*/`, this takes a huge amount of time to be processed. I guess that the best way to encode it is to separate the POS character from the others. Should one also consider to split all other values using multiFeat? How do you usually model morphological features in PAULA?

**Comment by thomaskrause** [comment]
I think is indeed best to split this up as much as possible. PAULA is agnostic to how to model morphological features, but the further processing is as you've seen dependent on the complexity of the annotation value. In general, regular expressions with prefixes work better than having a `.*` at the beginning, because ANNIS can narrow the search based on the prefix. Than all matches with the prefix have to be matched against the regular expression. Also, if there are fewer values possible in a tagset, the regular expression can be evaluated more easily and less storage is needed in the main memory implementation. If you use a combined morphological value that represents several dimensions, the number of combinations are larger and more distinct values have to be stored and searched.
